### PR TITLE
Validate config.yml when used with starter

### PR DESCRIPTION
**Pull request by DimedS**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description
This PR validates config.yml when used in conjunction with the --starter flag. It specifically prohibits the use of the "example_pipeline" or "tools" options in this scenario.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Review by ankatiyar — APPROVED** [review]
The error message is being shown correctly but the config files use `example_pipeline` and `tools` keys so I've left a suggestion, feel free to commit/modify the suggestion :)

File: kedro/framework/cli/starters.py [hunk 1]
```diff
@@ -730,17 +732,24 @@ def _validate_config_file_against_prompts(
...
     Args:
         config: The config as a dictionary
+        starter_alias: Starter alias if it was provided from CLI, otherwise None
 
     Raises:
         SystemExit: If the provided variables are not properly formatted.
     """
+    if starter_alias and ("tools" in config or "example_pipeline" in config):
+        raise KedroCliError(
+            "Cannot use the --starter flag with the --example and/or --tools options in config."
```
[the remark this answers is not in the document]
Comment (DimedS) [reply kedro/framework/cli/starters.py]: Thanks @ankatiyar , very good point! I committed it and updated test accordingly

**Review by SajidAlamQB — APPROVED** [review]
LGTM, thank you! @DimedS

**Review by SajidAlamQB — APPROVED** [review]

**Comment by DimedS** [comment]
Closes #3343

**Pull request merged**